### PR TITLE
Remove XFAIL test.opt files for passing tests #108

### PR DESCRIPTION
**Pull request by spanners**
* tests/attrib_size_object/test.out
* tests/attrib_size_object/test.opt
* tests/deferred_constant_use/test.opt
* tests/deferred_constant_use/test.out
* tests/enum_symbols_range_constraint_assignment/test.opt
* tests/enum_symbols_range_constraint_assignment/test.out
* tests/packages/test.opt
* tests/packages/test.out
* tests/record_extension/test.opt
* tests/record_extension/test.out
* tests/representation_clause_component_size_change/test.opt
* tests/representation_clause_component_size_change/test.out
* tests/size_attribute_clause/test.opt
* tests/size_attribute_clause/test.out
* tests/size_integer_and_objects/test.opt
* tests/size_integer_and_objects/test.out
Remove test.opt file and add the test.out output for each passing test

### CODE REVIEW

File: testsuite/gnat2goto/tests/attrib_size_object/test.out [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Standard_Output from gnat2goto object_size:
+----------At: Process_Declaration----------
+----------size clause not applied by the front-end----------
```
Comment (martin-cs) [inline testsuite/gnat2goto/tests/attrib_size_object/test.out]: Although the tests pass, this still has an unsupported feature so I don't think it should pass.

File: testsuite/gnat2goto/tests/record_extension/test.out [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Standard_Output from gnat2goto record_init:
+----------At: Do_Derived_Type_Definition----------
+----------record extension unsupported----------
```
Comment (martin-cs) [inline testsuite/gnat2goto/tests/record_extension/test.out]: Again, I don't think we should have unsupported features in the expected test results.

File: testsuite/gnat2goto/tests/representation_clause_component_size_change/test.out [hunk 3]
```diff
@@ -0,0 +1,15 @@
+Standard_Output from gnat2goto test:
+----------At: Process_Declaration----------
+----------Having component sizes be different from the size of their underlying type is currently not supported----------
```
Comment (martin-cs) [inline testsuite/gnat2goto/tests/representation_clause_component_size_change/test.out]: Again.

File: testsuite/gnat2goto/tests/size_attribute_clause/test.out [hunk 4]
```diff
@@ -0,0 +1,70 @@
+Standard_Output from gnat2goto check_size:
+----------At: Process_Declaration----------
+----------size clause not applied by the front-end----------
```
Comment (martin-cs) [inline testsuite/gnat2goto/tests/size_attribute_clause/test.out]: Another one.

File: testsuite/gnat2goto/tests/size_integer_and_objects/test.out [hunk 5]
```diff
@@ -1,19 +1,70 @@
-+===========================GNAT BUG DETECTED==============================+
-| GNU Ada (ada2goto) Assert_Failure tree_walk.adb:4673                     |
-| Error detected at check_size.adb:1:11                                    |
-| Please submit a bug report; see https://gcc.gnu.org/bugs/ .              |
-| Use a subject line meaningful to you and us to track the bug.            |
-| Include the entire contents of this bug box in the report.               |
-| Include the exact command that you entered.                              |
-| Also include sources listed below.                                       |
-+==========================================================================+
+Standard_Output from gnat2goto check_size:
```
Comment (martin-cs) [inline testsuite/gnat2goto/tests/size_integer_and_objects/test.out]: I think it is good to update this and remove the crash but I would prefer not having unsupported features in the golden, expected results.

**Comment by chrisr-diffblue** [comment]
@martin-cs - From a purist point of view, I'd agree that having no "missing feature" reports in the expected output of regression tests is probably the way to go, pragmatically having tests run and xfail doesn't provide us a *lot* of information. One strategy that might be worth exploring is to see if it's possible to split any of these tests so that the part of the test that would pass is in one test, and the part that gives the "missing feature" report is in a separate, XFAIL'ed test. That sort of gets the best of both worlds, but I also fully expect some of the tests won't be able to be split like that.

**Comment by chrisr-diffblue** [comment]
Either way, it might be worth splitting the current single commit into two commits - one commit containing just the tests that cleanly pass with no missing feature reports, and one commit for however the tests with missing feature reports is handled. That would mean the commit history would contain some documentation about the handling of the two different cases.

**Comment by spanners** [comment]
> and one commit for however the tests with missing feature reports is handled

is this 'however' decided? i.e. what should I do to handle the tests with missing feature report?

**Comment by martin-cs** [comment]
@spanners / @chrisr-diffblue  I am open to suggestions on the status of tests with unsupported features.  I feel like they shouldn't just pass.  Having them XFAIL is my preference I think but I can see an argument for passing but having the unsupported feature message in the expected results so that when it is fixed the test 'fails' and needs correcting.

Are there any existing tests that pass with unsupported features?

**Comment by martin-cs** [comment]
I also agree with the two commits approach.  Let's get the 'definitely working with all features supported' cases merged ASAP.

File: testsuite/gnat2goto/tests/size_integer_and_objects/test.out (hunk 5 above)
Comment (spanners) [reply testsuite/gnat2goto/tests/size_integer_and_objects/test.out]: Fixed

File: testsuite/gnat2goto/tests/size_attribute_clause/test.out (hunk 4 above)
Comment (spanners) [reply testsuite/gnat2goto/tests/size_attribute_clause/test.out]: Fixed

File: testsuite/gnat2goto/tests/representation_clause_component_size_change/test.out (hunk 3 above)
Comment (spanners) [reply testsuite/gnat2goto/tests/representation_clause_component_size_change/test.out]: Fixed

File: testsuite/gnat2goto/tests/record_extension/test.out (hunk 2 above)
Comment (spanners) [reply testsuite/gnat2goto/tests/record_extension/test.out]: Fixed

File: testsuite/gnat2goto/tests/attrib_size_object/test.out (hunk 1 above)
Comment (spanners) [reply testsuite/gnat2goto/tests/attrib_size_object/test.out]: Fixed

**Comment by spanners** [comment]
ce32931

> I also agree with the two commits approach. Let's get the 'definitely working with all features supported' cases merged ASAP.

I have created commit ce32931 which has the 'definitely working with all features supported' version.

File: testsuite/gnat2goto/tests/size_integer_and_objects/test.out [hunk 6]
```diff
@@ -1,19 +1,70 @@
...
-be reproduced with the set of files listed.
-Consider also -gnatd.n switch (see debug.adb).
-
-check_size.adb
-
-compilation abandoned
+[check_size.assertion.1] line 30 assertion Integer (Var_T_1) + Integer (Var_T_2) = 70: SUCCESS
+[check_size.assertion.2] line 34 assertion Var_Size_1 + Var_Size_2 = 30: SUCCESS
+[check_size.assertion.3] line 37 assertion Var_U8 + 1 = 0: SUCCESS
+VERIFICATION SUCCESSFUL
```
Comment (chrisr-diffblue) [inline testsuite/gnat2goto/tests/size_integer_and_objects/test.out]: Really minor nit pick - this file is ending without a newline. We typically make sure files do end with a newline character and Gituhub will highlight this as it has here.

Comment (spanners) [reply testsuite/gnat2goto/tests/size_integer_and_objects/test.out]: But this is the `test.out` file... I don't believe I have control over whether the file ends with a newline character or not. Do you know how I'd fix this (if possible)?

Comment (chrisr-diffblue) [reply testsuite/gnat2goto/tests/size_integer_and_objects/test.out]: I _think_ the diffing against the `test.out` expected output is flexible on the newline front, so you should be able to manually add the newline. Most of us here have our editors set to always add a newline at the end. Of course, if adding a newline causes the test matching to complain, then please consider me corrected :-)

**Comment by spanners** [comment]
I have no idea why the build is failing - I can't reproduce the diffs on my local machine when running the gnat2goto `python testsuite.py` command

**Comment by chrisr-diffblue** [comment]
@spanners - I just remebered that this sorting order may be affected by UNIX Locale settings - I think CI is running with `LC_ALL=posix`, so could you try running your local tests with `LC_ALL=posix`  and see if that changes the results?

**Comment by chrisr-diffblue** [comment]
Actually, we've just found a small bug in CBMC and how it sorts it's output. We'll get that fixed and update the submodule, and we'll ping you once thats in.

**Comment by chrisr-diffblue** [comment]
Once https://github.com/diffblue/gnat2goto/pull/330 is merged, hopefully the ordering issues will go away from the regression tests :-) Note that that PR includes a bump of the CBMC submodule, so after updating/rebasing your source tree you will need to ensure the submodule has updated, and then rebuild CBMC.

**Comment by thk123** [comment]
@spanners #330 is now merged which should fix your ordering problem on CI.

File: testsuite/gnat2goto/tests/address_clause/test.out [hunk 7]
```diff
@@ -18,6 +18,6 @@ N_Attribute_Definition_Clause "address" (Node_Id=2295) (source,analyzed)
  Entity = N_Defining_Identifier "var_addr_2" (Entity_Id=2282)
  Check_Address_Alignment = True
 
-[address_clause.assertion.2] line 12 assertion Var_Addr_1 + Var_Addr_2 = 3: SUCCESS
```
Comment (thk123) [inline testsuite/gnat2goto/tests/address_clause/test.out]: I'm confused why this change is showing up, when the exact same change is made in #330: https://github.com/diffblue/gnat2goto/pull/330/files#diff-b450147fb1fc8092f271746d6b36256e but I suppose it doesn't matter

Comment (thk123) [reply testsuite/gnat2goto/tests/address_clause/test.out]: I think it is simply that it is based on an old commit of master. If you rebase first, your commit will be a little smaller, but I don't mind either way

Comment (chrisr-diffblue) [reply testsuite/gnat2goto/tests/address_clause/test.out]: Yeah, I think this should be rebased so that the PR is against the newest commit of master - though possibly in Spanners defence, I did have a PR recently (in a different Github repo) where even though I'd rebased my code, force pushed it to the PR, etc - the PR was *still* showing the old version. In that instance, I changed the commit message slightly and then did another force push, and that seemed to be enough to get the PR showing the right thing. So @spanners - if you could double check you've done a fetch/rebase, and try pushing again, it would be good to have this PR showing the proper diff.

Comment (spanners) [reply testsuite/gnat2goto/tests/address_clause/test.out]: https://github.com/diffblue/gnat2goto/pull/327#discussion_r370011366 I *think* I've rebased now! Thanks for your patience.

File: testsuite/gnat2goto/tests/size_integer_and_objects/test.out (hunk 5 above)
Comment (chrisr-diffblue) [reply testsuite/gnat2goto/tests/size_integer_and_objects/test.out]: Looks like this test.out still contains the "unsupported feature" report (below)?

Comment (spanners) [reply testsuite/gnat2goto/tests/size_integer_and_objects/test.out]: Oops - missed that one. Should be fixed now by https://github.com/diffblue/gnat2goto/commit/2d836e0e56c80458063a182788a155e1eaf07e5f

**Comment by chrisr-diffblue** [comment]
@martin-cs I think this is now just waiting for your re-review, when you get a moment.
